src/Agent.Worker/Variables.cs

```diff
@@ -90,6 +149,135 @@ public bool TryGetValue(string name, out string val)
             _trace.Verbose($"Get '{name}' (not found)");
             return false;
         }
+
+        private void RecursivelyExpand(out List<string> warnings)
```

Comment:
This is the private method that is only used to recursively expand the job's variables. It is only called once in the lifetime of the worker - from the Variables ctor, from the job-level execution context's InitializeJob method, from JobRunner.